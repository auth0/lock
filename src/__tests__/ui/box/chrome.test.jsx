const getChrome = () => {
  return require('ui/box/chrome').default;
};

describe('Chrome', () => {
  let oldConsole;
  beforeEach(() => {
    oldConsole = console;
    global.console = { log: jest.fn() };
  });
  afterEach(() => {
    global.console = oldConsole;
  });
  it('`debug` calls console.log when __DEV__', () => {
    global.__DEV__ = true;
    const Chrome = getChrome();
    var c = new Chrome({
      contentProps: {
        model: {
          toJS: () => 'jsmodel'
        }
      }
    });

    c.debug();
    const { mock } = global.console.log;
    expect(mock.calls.length).toBe(1);
    expect(mock.calls[0][0]).toBe('jsmodel');
  });
  it('`debug` does not call console.log when !__DEV__', () => {
    global.__DEV__ = false;
    const Chrome = getChrome();
    var c = new Chrome({});

    c.debug();
    const { mock } = global.console.log;
    expect(mock.calls.length).toBe(0);
  });
});
