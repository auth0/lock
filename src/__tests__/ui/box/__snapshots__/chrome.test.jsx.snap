// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chrome can dislay all global messages together 1`] = `
<div
  className="auth0-lock-cred-pane auth0-lock-quiet"
>
  <div
    className="auth0-lock-cred-pane-internal-wrapper"
  >
    <div
      className="auth0-lock-content-wrapper"
    >
      <div
        data-__type="header"
        data-backHandler={undefined}
        data-backgroundColor="white"
        data-backgroundUrl="avatar"
        data-logoUrl="logo"
        data-name={undefined}
        data-title={undefined}
      />
      <div
        className="auth0-lock-content-body-wrapper"
      >
        <div>
          <div>
            <div
              data-__type="div"
              data-message={
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "This is a success message",
                    }
                  }
                />
              }
              data-scrollIntoView={true}
              data-type="success"
            />
            <div
              data-__type="div"
              data-message={
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "There is an error",
                    }
                  }
                />
              }
              data-scrollIntoView={true}
              data-type="error"
            />
            <div
              data-__type="div"
              data-message={
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "This is an information message",
                    }
                  }
                />
              }
              data-scrollIntoView={true}
              data-type="info"
            />
          </div>
        </div>
        <div
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <div
            data-__type="div"
            data-delay={550}
            data-onDidAppear={[Function]}
            data-onDidSlide={[Function]}
            data-onWillSlide={[Function]}
            data-reverse={false}
            data-transitionName=""
          >
            <div
              className="auth0-lock-view-content"
            >
              <div
                style={
                  Object {
                    "position": "relative",
                  }
                }
              >
                <div
                  className="auth0-lock-body-content"
                >
                  <div
                    className="auth0-lock-content"
                  >
                    <div
                      className="auth0-lock-form"
                    >
                      <div
                        data-__type="content"
                        data-focusSubmit={[Function]}
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      aria-label="Submit"
      className="auth0-lock-submit"
      disabled={false}
      name="submit"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "display": "block",
        }
      }
      type="submit"
    >
      <div
        className="auth0-loading-container"
      >
        <div
          className="auth0-loading"
        />
      </div>
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "<svg aria-hidden=\\"true\\" focusable=\\"false\\" width=\\"43px\\" height=\\"42px\\" viewBox=\\"0 0 43 42\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" xmlns:sketch=\\"http://www.bohemiancoding.com/sketch/ns\\"><g id=\\"Page-1\\" stroke=\\"none\\" stroke-width=\\"1\\" fill=\\"none\\" fill-rule=\\"evenodd\\" sketch:type=\\"MSPage\\"><g id=\\"Lock\\" sketch:type=\\"MSArtboardGroup\\" transform=\\"translate(-280.000000, -3592.000000)\\"><g id=\\"SMS\\" sketch:type=\\"MSLayerGroup\\" transform=\\"translate(153.000000, 3207.000000)\\"><g id=\\"Group\\" sketch:type=\\"MSShapeGroup\\"><g id=\\"Login\\" transform=\\"translate(0.000000, 369.000000)\\"><g id=\\"Btn\\"><g id=\\"Oval-302-+-Shape\\" transform=\\"translate(128.000000, 17.000000)\\"><circle id=\\"Oval-302\\" stroke=\\"#FFFFFF\\" stroke-width=\\"2\\" cx=\\"20.5\\" cy=\\"20\\" r=\\"20\\"></circle><path d=\\"M17.8,15.4 L19.2,14 L25.2,20 L19.2,26 L17.8,24.6 L22.4,20 L17.8,15.4 Z\\" id=\\"Shape\\" fill=\\"#FFFFFF\\"></path></g></g></g></g></g></g></g></svg>",
          }
        }
      />
    </button>
  </div>
</div>
`;

exports[`Chrome renders correctly when there is a global information message 1`] = `
<div
  className="auth0-lock-cred-pane auth0-lock-quiet"
>
  <div
    className="auth0-lock-cred-pane-internal-wrapper"
  >
    <div
      className="auth0-lock-content-wrapper"
    >
      <div
        data-__type="header"
        data-backHandler={undefined}
        data-backgroundColor="white"
        data-backgroundUrl="avatar"
        data-logoUrl="logo"
        data-name={undefined}
        data-title={undefined}
      />
      <div
        className="auth0-lock-content-body-wrapper"
      >
        <div>
          <div>
            <div
              data-__type="div"
              data-message={
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "This is an information message",
                    }
                  }
                />
              }
              data-scrollIntoView={true}
              data-type="info"
            />
          </div>
        </div>
        <div
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <div
            data-__type="div"
            data-delay={550}
            data-onDidAppear={[Function]}
            data-onDidSlide={[Function]}
            data-onWillSlide={[Function]}
            data-reverse={false}
            data-transitionName=""
          >
            <div
              className="auth0-lock-view-content"
            >
              <div
                style={
                  Object {
                    "position": "relative",
                  }
                }
              >
                <div
                  className="auth0-lock-body-content"
                >
                  <div
                    className="auth0-lock-content"
                  >
                    <div
                      className="auth0-lock-form"
                    >
                      <div
                        data-__type="content"
                        data-focusSubmit={[Function]}
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      aria-label="Submit"
      className="auth0-lock-submit"
      disabled={false}
      name="submit"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "display": "block",
        }
      }
      type="submit"
    >
      <div
        className="auth0-loading-container"
      >
        <div
          className="auth0-loading"
        />
      </div>
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "<svg aria-hidden=\\"true\\" focusable=\\"false\\" width=\\"43px\\" height=\\"42px\\" viewBox=\\"0 0 43 42\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" xmlns:sketch=\\"http://www.bohemiancoding.com/sketch/ns\\"><g id=\\"Page-1\\" stroke=\\"none\\" stroke-width=\\"1\\" fill=\\"none\\" fill-rule=\\"evenodd\\" sketch:type=\\"MSPage\\"><g id=\\"Lock\\" sketch:type=\\"MSArtboardGroup\\" transform=\\"translate(-280.000000, -3592.000000)\\"><g id=\\"SMS\\" sketch:type=\\"MSLayerGroup\\" transform=\\"translate(153.000000, 3207.000000)\\"><g id=\\"Group\\" sketch:type=\\"MSShapeGroup\\"><g id=\\"Login\\" transform=\\"translate(0.000000, 369.000000)\\"><g id=\\"Btn\\"><g id=\\"Oval-302-+-Shape\\" transform=\\"translate(128.000000, 17.000000)\\"><circle id=\\"Oval-302\\" stroke=\\"#FFFFFF\\" stroke-width=\\"2\\" cx=\\"20.5\\" cy=\\"20\\" r=\\"20\\"></circle><path d=\\"M17.8,15.4 L19.2,14 L25.2,20 L19.2,26 L17.8,24.6 L22.4,20 L17.8,15.4 Z\\" id=\\"Shape\\" fill=\\"#FFFFFF\\"></path></g></g></g></g></g></g></g></svg>",
          }
        }
      />
    </button>
  </div>
</div>
`;

exports[`Chrome renders correctly when there is a global message 1`] = `
<div
  className="auth0-lock-cred-pane auth0-lock-quiet"
>
  <div
    className="auth0-lock-cred-pane-internal-wrapper"
  >
    <div
      className="auth0-lock-content-wrapper"
    >
      <div
        data-__type="header"
        data-backHandler={undefined}
        data-backgroundColor="white"
        data-backgroundUrl="avatar"
        data-logoUrl="logo"
        data-name={undefined}
        data-title={undefined}
      />
      <div
        className="auth0-lock-content-body-wrapper"
      >
        <div>
          <div>
            <div
              data-__type="div"
              data-message={
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "There is an error",
                    }
                  }
                />
              }
              data-scrollIntoView={true}
              data-type="error"
            />
          </div>
        </div>
        <div
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <div
            data-__type="div"
            data-delay={550}
            data-onDidAppear={[Function]}
            data-onDidSlide={[Function]}
            data-onWillSlide={[Function]}
            data-reverse={false}
            data-transitionName=""
          >
            <div
              className="auth0-lock-view-content"
            >
              <div
                style={
                  Object {
                    "position": "relative",
                  }
                }
              >
                <div
                  className="auth0-lock-body-content"
                >
                  <div
                    className="auth0-lock-content"
                  >
                    <div
                      className="auth0-lock-form"
                    >
                      <div
                        data-__type="content"
                        data-focusSubmit={[Function]}
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      aria-label="Submit"
      className="auth0-lock-submit"
      disabled={false}
      name="submit"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "display": "block",
        }
      }
      type="submit"
    >
      <div
        className="auth0-loading-container"
      >
        <div
          className="auth0-loading"
        />
      </div>
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "<svg aria-hidden=\\"true\\" focusable=\\"false\\" width=\\"43px\\" height=\\"42px\\" viewBox=\\"0 0 43 42\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" xmlns:sketch=\\"http://www.bohemiancoding.com/sketch/ns\\"><g id=\\"Page-1\\" stroke=\\"none\\" stroke-width=\\"1\\" fill=\\"none\\" fill-rule=\\"evenodd\\" sketch:type=\\"MSPage\\"><g id=\\"Lock\\" sketch:type=\\"MSArtboardGroup\\" transform=\\"translate(-280.000000, -3592.000000)\\"><g id=\\"SMS\\" sketch:type=\\"MSLayerGroup\\" transform=\\"translate(153.000000, 3207.000000)\\"><g id=\\"Group\\" sketch:type=\\"MSShapeGroup\\"><g id=\\"Login\\" transform=\\"translate(0.000000, 369.000000)\\"><g id=\\"Btn\\"><g id=\\"Oval-302-+-Shape\\" transform=\\"translate(128.000000, 17.000000)\\"><circle id=\\"Oval-302\\" stroke=\\"#FFFFFF\\" stroke-width=\\"2\\" cx=\\"20.5\\" cy=\\"20\\" r=\\"20\\"></circle><path d=\\"M17.8,15.4 L19.2,14 L25.2,20 L19.2,26 L17.8,24.6 L22.4,20 L17.8,15.4 Z\\" id=\\"Shape\\" fill=\\"#FFFFFF\\"></path></g></g></g></g></g></g></g></svg>",
          }
        }
      />
    </button>
  </div>
</div>
`;

exports[`Chrome renders correctly when there is a global success message 1`] = `
<div
  className="auth0-lock-cred-pane auth0-lock-quiet"
>
  <div
    className="auth0-lock-cred-pane-internal-wrapper"
  >
    <div
      className="auth0-lock-content-wrapper"
    >
      <div
        data-__type="header"
        data-backHandler={undefined}
        data-backgroundColor="white"
        data-backgroundUrl="avatar"
        data-logoUrl="logo"
        data-name={undefined}
        data-title={undefined}
      />
      <div
        className="auth0-lock-content-body-wrapper"
      >
        <div>
          <div>
            <div
              data-__type="div"
              data-message={
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "This is a success message",
                    }
                  }
                />
              }
              data-scrollIntoView={true}
              data-type="success"
            />
          </div>
        </div>
        <div
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <div
            data-__type="div"
            data-delay={550}
            data-onDidAppear={[Function]}
            data-onDidSlide={[Function]}
            data-onWillSlide={[Function]}
            data-reverse={false}
            data-transitionName=""
          >
            <div
              className="auth0-lock-view-content"
            >
              <div
                style={
                  Object {
                    "position": "relative",
                  }
                }
              >
                <div
                  className="auth0-lock-body-content"
                >
                  <div
                    className="auth0-lock-content"
                  >
                    <div
                      className="auth0-lock-form"
                    >
                      <div
                        data-__type="content"
                        data-focusSubmit={[Function]}
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      aria-label="Submit"
      className="auth0-lock-submit"
      disabled={false}
      name="submit"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "display": "block",
        }
      }
      type="submit"
    >
      <div
        className="auth0-loading-container"
      >
        <div
          className="auth0-loading"
        />
      </div>
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "<svg aria-hidden=\\"true\\" focusable=\\"false\\" width=\\"43px\\" height=\\"42px\\" viewBox=\\"0 0 43 42\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" xmlns:sketch=\\"http://www.bohemiancoding.com/sketch/ns\\"><g id=\\"Page-1\\" stroke=\\"none\\" stroke-width=\\"1\\" fill=\\"none\\" fill-rule=\\"evenodd\\" sketch:type=\\"MSPage\\"><g id=\\"Lock\\" sketch:type=\\"MSArtboardGroup\\" transform=\\"translate(-280.000000, -3592.000000)\\"><g id=\\"SMS\\" sketch:type=\\"MSLayerGroup\\" transform=\\"translate(153.000000, 3207.000000)\\"><g id=\\"Group\\" sketch:type=\\"MSShapeGroup\\"><g id=\\"Login\\" transform=\\"translate(0.000000, 369.000000)\\"><g id=\\"Btn\\"><g id=\\"Oval-302-+-Shape\\" transform=\\"translate(128.000000, 17.000000)\\"><circle id=\\"Oval-302\\" stroke=\\"#FFFFFF\\" stroke-width=\\"2\\" cx=\\"20.5\\" cy=\\"20\\" r=\\"20\\"></circle><path d=\\"M17.8,15.4 L19.2,14 L25.2,20 L19.2,26 L17.8,24.6 L22.4,20 L17.8,15.4 Z\\" id=\\"Shape\\" fill=\\"#FFFFFF\\"></path></g></g></g></g></g></g></g></svg>",
          }
        }
      />
    </button>
  </div>
</div>
`;

exports[`Chrome renders correctly with basic props 1`] = `
<div
  className="auth0-lock-cred-pane auth0-lock-quiet"
>
  <div
    className="auth0-lock-cred-pane-internal-wrapper"
  >
    <div
      className="auth0-lock-content-wrapper"
    >
      <div
        data-__type="header"
        data-backHandler={undefined}
        data-backgroundColor="white"
        data-backgroundUrl="avatar"
        data-logoUrl="logo"
        data-name={undefined}
        data-title={undefined}
      />
      <div
        className="auth0-lock-content-body-wrapper"
      >
        <div>
          <div />
        </div>
        <div
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <div
            data-__type="div"
            data-delay={550}
            data-onDidAppear={[Function]}
            data-onDidSlide={[Function]}
            data-onWillSlide={[Function]}
            data-reverse={false}
            data-transitionName=""
          >
            <div
              className="auth0-lock-view-content"
            >
              <div
                style={
                  Object {
                    "position": "relative",
                  }
                }
              >
                <div
                  className="auth0-lock-body-content"
                >
                  <div
                    className="auth0-lock-content"
                  >
                    <div
                      className="auth0-lock-form"
                    >
                      <div
                        data-__type="content"
                        data-focusSubmit={[Function]}
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      aria-label="Submit"
      className="auth0-lock-submit"
      disabled={false}
      name="submit"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "display": "block",
        }
      }
      type="submit"
    >
      <div
        className="auth0-loading-container"
      >
        <div
          className="auth0-loading"
        />
      </div>
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "<svg aria-hidden=\\"true\\" focusable=\\"false\\" width=\\"43px\\" height=\\"42px\\" viewBox=\\"0 0 43 42\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" xmlns:sketch=\\"http://www.bohemiancoding.com/sketch/ns\\"><g id=\\"Page-1\\" stroke=\\"none\\" stroke-width=\\"1\\" fill=\\"none\\" fill-rule=\\"evenodd\\" sketch:type=\\"MSPage\\"><g id=\\"Lock\\" sketch:type=\\"MSArtboardGroup\\" transform=\\"translate(-280.000000, -3592.000000)\\"><g id=\\"SMS\\" sketch:type=\\"MSLayerGroup\\" transform=\\"translate(153.000000, 3207.000000)\\"><g id=\\"Group\\" sketch:type=\\"MSShapeGroup\\"><g id=\\"Login\\" transform=\\"translate(0.000000, 369.000000)\\"><g id=\\"Btn\\"><g id=\\"Oval-302-+-Shape\\" transform=\\"translate(128.000000, 17.000000)\\"><circle id=\\"Oval-302\\" stroke=\\"#FFFFFF\\" stroke-width=\\"2\\" cx=\\"20.5\\" cy=\\"20\\" r=\\"20\\"></circle><path d=\\"M17.8,15.4 L19.2,14 L25.2,20 L19.2,26 L17.8,24.6 L22.4,20 L17.8,15.4 Z\\" id=\\"Shape\\" fill=\\"#FFFFFF\\"></path></g></g></g></g></g></g></g></svg>",
          }
        }
      />
    </button>
  </div>
</div>
`;
